### PR TITLE
Regression testing in torchtune.

### PR DESCRIPTION
**Pull request by krammnic**
#### Context
What is the purpose of this PR? Is it to
- [X] add a new feature
- [ ] fix a bug
- [X] update tests and/or documentation
- [ ] other (please add here)

Please link to any issues this PR addresses.

#### Changelog
What are the changes made in this PR?
* #1306 

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [ ] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [ ] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Here is a [docstring example](https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285)
and a [tutorial example](https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models)

- [ ] I did not change any public API
- [ ] I have added an example to docs or docstrings

### CODE REVIEW

**Comment by krammnic** [comment]
cc @ebsmothers

File: recipes/configs/qwen2_5/evaluation.yaml [hunk 1]
```diff
@@ -7,13 +7,14 @@ output_dir: ./ # Not needed
 
 # Model Arguments
 model:
-  _component_: torchtune.models.qwen2_5.qwen2_5_0_5b
+  _component_: torchtune.models.qwen2_5.qwen2_5_3b
 
 checkpointer:
   _component_: torchtune.training.FullModelHFCheckpointer
   checkpoint_dir: /tmp/Qwen2.5-0_5B-Instruct
```
Comment (maximegmd) [inline recipes/configs/qwen2_5/evaluation.yaml:14]: Path should also be updated to match the model version

File: tests/regression_tests/test_qwen2_5_3b.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+@gpu_test(gpu_count=2)
+class TestFull3BDistributedFinetuneDPO:
+    @pytest.mark.slow_integration_test
+    def test_finetune_and_eval(self, tmpdir, caplog, monkeypatch):
+        ckpt_path = Path(CKPT_MODEL_PATHS[CKPT])
+
+        # Run on prod Full DPO config but with only 20 steps for now
+        ft_cmd = f"""
+        tune run --nnodes 1 --nproc_per_node 2 full_dpo_distributed
+            --config llama3_1/8B_full_dpo \
```
Comment (maximegmd) [inline tests/regression_tests/test_qwen2_5_3b.py:138]: Config should be qwen dpo?

File: recipes/configs/qwen2_5/evaluation.yaml (hunk 1 above)
Comment (krammnic) [reply recipes/configs/qwen2_5/evaluation.yaml:14]: Good catch, we even don't need to change this config

File: tests/regression_tests/test_qwen2_5_3b.py (hunk 2 above)
Comment (krammnic) [reply tests/regression_tests/test_qwen2_5_3b.py:138]: We don't have such, do we?

Comment (maximegmd) [reply tests/regression_tests/test_qwen2_5_3b.py:138]: Yeah but we also mention qwen 2.5 here
